lib/blockchain.go

```diff
@@ -75,6 +75,13 @@ const (
 	StatusBitcoinHeaderValidateFailed // Deprecated
 )
 
+type CommittedBlockStatus uint8
+
+const (
+	COMMITTED   CommittedBlockStatus = 0
+	UNCOMMITTED CommittedBlockStatus = 1
+)
```

Comment:
As I read through this, I had the following thoughts:
* Why not overload BlockStatus to add a StatusBlockCommitted field?
* If we did this, then we wouldn't need the uncommittedBlockMap. Instead, blocks that are in the blockIndex are implicitly uncommitted if they have StatusBlockValidated but not StatusBlockCommitted.
* If we did the above, the only thing you'd need to get the "block tip" is the uncommitted tip. To get this, I was thinking we could just change the "best chain tip" to be the uncommitted tip. Then if you want the committed tip, you use the blockIndex to wlak back from the "best chain tip" to the first block with StatusBlockCommitted.

Above is the most lightweight way to store blocks in PoS but obviously haven't thought through all the implications.

Reply:
Also I don't think you'd have to add anything to the BlockNode structure if you did the above ^

Reply:
This is a work in progress in https://github.com/deso-protocol/core/pull/742. Marking as resolved since it's implemented there